### PR TITLE
LPS-121673 Remove unused submit event handler from dialog modal

### PR DESCRIPTION
**Pull request by albinojunior**
**[BUG](https://issues.liferay.com/browse/LPS-121673):** Using any dialog modal created by the `Liferay.Util.openWindow`, after click on the save/submit button the cancel button doesn't work as proposed and the dialog doesn't close.

**FIX:** This PR removes the events detachment when submit the form inside the dialog modal, and keep only the cancel event detachment allowing the user submit many times and close the modal when the cancel button is clicked.

**TEST STEPS:**
1. Go to Site Menu > Site Builder > Pages > Create a Widget page and set it as "1 Column"
2. Go to the page created and add a Calendar widget
3. Click on the "arrow" icon next to one of the calendars > Calendar Settings
4. Click on "Save"
5. Click on "Cancel"

### CODE REVIEW

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Hey @markocikos, are we good with this?

For reference, here's the initial [Slack Conversation](https://liferay.slack.com/archives/CNBG06JS3/p1607688702399900)

@albinojunior, while we finish taking a look at this, could you please take a look at the following guidelines on how our PR workflow works and update this PR accordingly?
- [Frontend Infrastructure Team code review process](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/code_review.md)
- [Pull Request Etiquette](https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/pull_requests.md)

In summary we at least expect in a PR:
- A descriptive **title** other than the LPS number (✅ )
- A clear **description** of what is in the PR including the expectations you have for it (us forward it, gather feedback...) and things you tried, why you think this is the right fix... (🔴 )
- A simple **test plan** with reproducible steps so reviewers can easily decide how much testing the PR needs (🔴 )

We greatly appreciate all your help in making this workflow better for everyone! ❤️

**Review by markocikos — APPROVED** [review]
LGTM!

I can't think of a reason why we would want to always detach cancel listener on submit. Maybe in old times modals use to always close on save, but this is not a common workflow nowadays.

Even if we have the variant where modal closes on submit, such as [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance.jsp#L313-L319), it is not a problem. Legacy modals can be:
1. hidden and reopened, if `destroyOnHide` is `false`. In that case, we don't want to get rid of the listener. - or -
2. removed from DOM, if `destroyOnHide` is `true`. In that case, cancel listener will be removed, as it is defined on `iframeBody`

Finally, if we do break some edge case, it's (probably) not a big deal. `openWindow`s is a legacy util, should be replaced with `openModal`

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
@john-co Please take a look at failure, it looks unrelated.

**Comment by jbalsas** [comment]
Error in common with other PRs, forwarding manually

**Comment by jbalsas** [comment]
Forwarded at https://github.com/brianchandotcom/liferay-portal/pull/97048
